### PR TITLE
Add documentation for missing-any-param-doc, missing-kwon, subprocess-popen-preexec-fn and typevar-name-incorrect-variance

### PR DESCRIPTION
**Pull request by ollie-iterators**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Add documentation for missing-any-param-doc, missing-kwon, subprocess-popen-preexec-fn and typevar-name-incorrect-variance

Refs #5953 

Closes #XXXX

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Could you remove typevar-name-incorrect-variance from this PR, it need to be documented by one of the core maintainer ?

**Comment by ollie-iterators** [comment]
> Could you remove typevar-name-incorrect-variance from this PR, it need to be documented by one of the core maintainer ?

I can do that later today

**Comment by ollie-iterators** [comment]
Sorry for the extra commits. I added the changes from my phone

File: doc/data/messages/t/typevar-name-incorrect-variance/details.rst [hunk 1]
```diff
@@ -1 +0,0 @@
-You can help us make the doc better `by contributing <https://github.com/PyCQA/pylint/issues/5953>`_ !
```
[the remark this answers is not in the document]
Comment (ollie-iterators) [reply doc/data/messages/t/typevar-name-incorrect-variance/details.rst:1]: Should I remove it or should @DanielNoord fix it?

[the remark this answers is not in the document]
Comment (ollie-iterators) [reply doc/data/messages/t/typevar-name-incorrect-variance/details.rst:1]: Removed truncated-format-string from the pull request

[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply doc/data/messages/t/typevar-name-incorrect-variance/details.rst:1]: It was actually 'typevar-name-incorrect-variance' that should be removed, but I'm happy with PR with less message documented at once personally :)

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
All right, I think you removed the wrong message again but we can make the message better/more complete later anyway 😄

**Pull request merged**